server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -501,8 +511,47 @@ private static SnapshotsInRepo sortSnapshots(
         final @Nullable GetSnapshotsRequest.After after,
         final int offset,
         final int size,
-        final SortOrder order
+        final SortOrder order,
+        final String[] slmPolicies
     ) {
+        final List<SnapshotInfo> filteredSnapshotInfos;
```

Comment:
Could we make the filtering here a separate method instead? It looks like there is at least one usage where this is not needed and it is a bit odd to include into the sort method which is also already long enough as it stands. Perhaps add a `sortAndFilter` that calls both too.

Reply:
Sounds good, made it into 2 methods and kept the old sort method for those spots that have no filtering :)